Fix K8s/OpenShift deployment when using management interface+https

Sgitario opened the pull request:
Note that these changes will move the management ssl configuration to built-time configuration. 

Moreover, the logic to select the HTTPS schema mimics to the logic in VertxHttpRecorder.initializeMainHttpServer method.

Fix https://github.com/quarkusio/quarkus/issues/32225

gsmet commented:
Pinging @geoand too. Moving securty-related config to build time looks a bit impractical to me.

geoand commented:
Unfortunately with all the stuff that is going on, I won't be able to look at this for at least 10 days

cescoffier commented:
Let's discuss this after devoxx greece. While I agreed on reducing the flexibility, it can become problematic. Especially because I was also thinking of reducing the TLS configuration requirement (if you configure a cert, it's enabled). This approach may prevent these improvements.

Sgitario commented:
> Let's discuss this after devoxx greece. While I agreed on reducing the flexibility, it can become problematic. Especially because I was also thinking of reducing the TLS configuration requirement (if you configure a cert, it's enabled). This approach may prevent these improvements.

Sure. All these arguments go into the direction of adding a new property `quarkus.management.ssl.certificate.enabled=true` to users enable it on. 
+1 Let's postpone it after the Devoxx greece.

Sgitario commented on extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java:
```diff
@@ -8,12 +8,22 @@
         return name;
     }
+
+    public String getSchema() {
```
ops! PR updated.

cescoffier commented on extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java:
```diff
@@ -466,4 +477,17 @@ public String updateApiUrl(String original, String healthPath) {
     }
+
+    private static String selectSchemeForManagement() {
+        Config config = ConfigProvider.getConfig();
```
I would add a comment saying this is a best-effort approach. 
We also discussed a warning with @iocanel

Sgitario replied:
I understood that it should be INFO messages, not warnings. If so, these will be covered in a separate task (see the third point in epic https://github.com/quarkusio/quarkus/issues/33307).
I'll add a comment about the best-effort approach.

Sgitario commented:
@iocanel can you have a look at this pull request please? Thanks!

iocanel commented on extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java:
```diff
@@ -36,6 +36,12 @@ public class ProbeConfig {
+     * The schema of the {@literal HTTP get} action. Can be either "HTTP" or "HTTPS".
+     */
+    @ConfigItem
+    Optional<String> httpActionSchema;
```
Is this correct? Or it was meant to be `httpActionScheme`?

Sgitario replied:
Ops, thanks for spotting! 
PR updated

iocanel reviewed (approved):
lgtm

Sgitario committed:
Fix K8s/OpenShift deployment when using management interface+https

Note that these changes will move the management ssl configuration to built-time configuration. 

Moreover, the logic to select the HTTPS schema mimics to the logic in VertxHttpRecorder.initializeMainHttpServer method.

Fix https://github.com/quarkusio/quarkus/issues/32225

Sgitario commented:
I'm not sure if the failing tests are related or not. I will take a look asap.

Sgitario commented:
I tried the test that failed locally, and it worked for me. 
@geoand @gsmet do we know if these tests are flaky?

geoand commented:
It might be a case of us needing to increase the timeouts for the native tests

Sgitario commented:
CI is green now. Merging!
